File: src/main/java/no/novari/flyt/authorization/client/sourceapplications/model/AltinnSourceApplication.java
package no.novari.flyt.authorization.client.sourceapplications.model;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class AltinnSourceApplication extends BaseSourceApplication {
    public AltinnSourceApplication(
            @Value("${fint.flyt.altinn.sso.client-id:#{null}}") String clientId,
            @Value("${fint.flyt.altinn.available:true}") boolean available
    ) {
        super(5L, "Altinn", clientId, available);
    }
}

File: src/main/kotlin/no/novari/flyt/authorization/client/sourceapplications/model/AltinnSourceApplication.kt
package no.novari.flyt.authorization.client.sourceapplications.model

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class AltinnSourceApplication(
    @Value("\${fint.flyt.altinn.sso.client-id:#{null}}")
    clientId: String?,
    @Value("\${fint.flyt.altinn.available:true}")
    available: Boolean,
) : BaseSourceApplication(
        id = 5L,
        displayName = "Altinn",
        clientId = clientId,
        available = available,
    )
